clients/drcachesim/scheduler/scheduler_impl.cpp

```diff
@@ -2861,8 +2862,8 @@ scheduler_impl_tmpl_t<RecordType, ReaderType>::next_record(output_ordinal_t outp
     }
     VPRINT(this, 4, "next_record[%d]: from %d @%" PRId64 ": ", output, input->index,
            cur_time);
-    update_next_record(output, record);
     VDO(this, 4, print_record(record););
+    update_next_record(output, record);
```

Comment:
Wouldn't we prefer to see the final record, i.e., update first then print?

Reply:
Right. Done.